docs/spec/intermediate_representation.md

```diff
@@ -476,6 +476,27 @@ for details on the field definition format.
 
 Note, the difference between safe and unsafe arguments are explained in the docs of [conjure-java-runtime](https://github.com/palantir/conjure-java-runtime#error-propagation).
 
+## Extensions
+
+The "extensions" section is a map of named extensions. The structure of each extension is user defined and
+any consumer of an extension should treat its presence as optional and be tolerant of unknown fields. Extensions must
+not modify the semantics of definitions and are intended only for conveying metadata about the definition.
+
+Example extension definition:
+```json
+{
+  "extensions": {
+    "product-dependencies": [
```

Comment:
should this be sls-recommended-product-depndencies to match what we put in jar manifests?

Reply:
sure, I'll update https://github.com/palantir/gradle-conjure/pull/381 to use `sls-recommended-product-dependencies`